### DISCUSSION TITLE
AWS ECS: Firefox can't establish a connection to the server" Error in Email Reports

### DISCUSSION THREAD

**Discussion by dfoley84** [apache/superset#26999] [Q&A / Help]
### Bug description

I am running Superset within an ECS Cluster, which I'm trying to set up email reporting within Superset, I'm currently using the Docker 3.1.0 non-dev container.

```
#Installing Webdriver for Reporting 
RUN apt-get update && \
    apt-get install --no-install-recommends -y firefox-esr

ENV GECKODRIVER_VERSION=0.34.0
RUN wget -q https://github.com/mozilla/geckodriver/releases/download/v${GECKODRIVER_VERSION}/geckodriver-v${GECKODRIVER_VERSION}-linux64.tar.gz && \
    tar -x geckodriver -zf geckodriver-v${GECKODRIVER_VERSION}-linux64.tar.gz -O > /usr/bin/geckodriver && \
    chmod 755 /usr/bin/geckodriver && \
    rm geckodriver-v${GECKODRIVER_VERSION}-linux64.tar.gz
```

My SMTP Configuration is the following; 

```
WEBDRIVER_BASEURL = 'http://0.0.0.0:8088'
WEBDRIVER_BASEURL_USER_FRIENDLY = WEBDRIVER_BASEURL

#Slack Config
SLACK_API_TOKEN =  get_env_variable("SLACK_TOKEN")

#SMTP Config
SMTP_HOST = get_env_variable("SMTP_HOST")
SMTP_PORT = get_env_variable("SMTP_PORT")
SMTP_STARTTLS = True
SMTP_SSL_SERVER_AUTH = True
SMTP_SSL = True
SMTP_USER = get_env_variable("SMTP_USERNAME")
SMTP_PASSWORD = get_env_variable("SMTP_PASSWORD")
SMTP_MAIL_FROM  = get_env_variable("SMTP_MAILFROM")
EMAIL_REPORTS_SUBJECT_PREFIX = "[Superset]"
```

### How to reproduce the bug

When I try and setup a report I'm getting the following 

`Failed taking a screenshot Message: Reached error page: about:neterror?e=connectionFailure&u=http%3A//0.0.0.0%3A8088/login/&c=UTF-8&d=Firefox%20can%E2%80%99t%20establish%20a%20connection%20to%20the%20server%20at%200.0.0.0%3A8088.
 `

Within the Logs I'm getting the following; 

`[2024-02-02 16:34:31,852: WARNING/ForkPoolWorker-2] SupersetError(message='[Errno 110] Connection timed out', error_type=<SupersetErrorType.REPORT_NOTIFICATION_ERROR: 'REPORT_NOTIFICATION_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra=None) A downstream exception occurred while generating a report: 61da970b-0503-4e62-a58e-bc07e2f596b6. Failed generating csv <urlopen error [Errno 111] Connection refused> Traceback (most recent call last): File "/usr/local/lib/python3.9/urllib/request.py", line 1346, in do_open h.request(req.get_method(), req.selector, req.data, headers, File "/usr/local/lib/python3.9/http/client.py", line 1285, in request self._send_request(method, url, body, headers, encode_chunked) File "/usr/local/lib/python3.9/http/client.py", line 1331, in _send_request self.endheaders(body, encode_chunked=encode_chunked) File "/usr/local/lib/python3.9/http/client.py", line 1280, in endheaders self._send_output(message_body, encode_chunked=encode_chunked) File "/usr/local/lib/python3.9/http/client.py", line 1040, in _send_output self.send(msg) File "/usr/local/lib/python3.9/http/client.py", line 980, in send self.connect() File "/usr/local/lib/python3.9/http/client.py", line 946, in connect self.sock = self._create_connection( File "/usr/local/lib/python3.9/socket.py", line 844, in create_connection raise err File "/usr/local/lib/python3.9/socket.py", line 832, in create_connection sock.connect(sa) ConnectionRefusedError: [Errno 111] Connection refused During handling of the above exception, another exception occurred: Traceback (most recent call last): File "/app/superset/commands/report/execute.py", line 258, in _get_csv_data csv_data = get_chart_csv_data(chart_url=url, auth_cookies=auth_cookies) File "/app/superset/utils/csv.py", line 92, in get_chart_csv_data response = opener.open(chart_url) File "/usr/local/lib/python3.9/urllib/request.py", line 517, in open response = self._open(req, data) File "/usr/local/lib/python3.9/urllib/request.py", line 534, in _open result = self._call_chain(self.handle_open, protocol, protocol + File "/usr/local/lib/python3.9/urllib/request.py", line 494, in _call_chain result = func(*args) File "/usr/local/lib/python3.9/urllib/request.py", line 1375, in http_open return self.do_open(http.client.HTTPConnection, req) File "/usr/local/lib/python3.9/urllib/request.py", line 1349, in do_open raise URLError(err) urllib.error.URLError: <urlopen error [Errno 111] Connection refused> The above exception was the direct cause of the following exception: Traceback (most recent call last): File "/app/superset/tasks/scheduler.py", line 94, in execute AsyncExecuteReportScheduleCommand( File "/app/superset/commands/report/execute.py", line 729, in run raise ex File "/app/superset/commands/report/execute.py", line 725, in run ReportScheduleStateMachine( File "/app/superset/commands/report/execute.py", line 684, in run state_cls( File "/app/superset/commands/report/execute.py", line 586, in next raise first_ex File "/app/superset/commands/report/execute.py", line 555, in next self.send() File "/app/superset/commands/report/execute.py", line 452, in send notification_content = self._get_notification_content() File "/app/superset/commands/report/execute.py", line 364, in _get_notification_content csv_data = self._get_csv_data() File "/app/superset/commands/report/execute.py", line 262, in _get_csv_data raise ReportScheduleCsvFailedError( superset.commands.report.exceptions.ReportScheduleCsvFailedError: Failed generating csv <urlopen error [Errno 111] Connection refused>`

### Superset version

3.1.0

### Python version

3.9

### Node version

Not applicable

### Browser

Firefox

### Additional context

I've tried using 

0.0.0.0, localhost, 127.0.0.1 and the Public Hostname for the Superset running within ECS all have the same issue as above

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by sfirke** [comment]
Hello, it appears this is a question of how to configure Reports on AWS ECS, rather than a bug in Superset's codebase.  I'm going to convert it to a discussion to keep the bug backlog manageable, then I'll respond there.

**Comment by sfirke** [comment]
This is probably the hardest part of Superset to troubleshoot.  There are some steps listed here - I'm linking to the particular one I think you should try: https://superset.apache.org/docs/configuration/alerts-reports/#browse-to-your-report-from-the-worker.  You can also try asking on the Superset Slack (first search the existing posts!) to see if anyone has tips particularly for AWS ECS.
